pkg/modprovider/module.go

```diff
@@ -299,9 +299,10 @@ func (h *moduleHandler) initializationError(outputs resource.PropertyMap, reason
 	contract.AssertNoErrorf(err, "plugin.MarshalProperties failed")
 
 	detail := pulumirpc.ErrorResourceInitFailed{
-		Id:         moduleStateResourceID,
-		Properties: props,
-		Reasons:    reasons,
+		Id:                  moduleStateResourceID,
+		Properties:          props,
+		Reasons:             reasons,
+		RefreshBeforeUpdate: true,
```

Comment:
Setting `RefreshBeforeUpdate` for the partial failure case, like what we do for the success case for `Create`, `Read`, and `Update`, e.g.:

https://github.com/pulumi/pulumi-terraform-module/blob/79259e03b2ca1a04e9f5f16fe6d1b558019951ad/pkg/modprovider/module.go#L387-L391